File: Diagram/ProbabilisticTransitionLine.vb
'************************************************************************************
' ST-Sim: A .NET class library for developing state-and-transition simulation models.
'
' Copyright © 2007-2017 Apex Resource Management Solution Ltd. (ApexRMS). All rights reserved.
'
'************************************************************************************

Imports System.Drawing

Class ProbabilisticTransitionLine
    Inherits TransitionDiagramLine

    Private m_TransitionTypeId As Integer
    Private m_TransitionGroups As New List(Of Integer)

    Public Sub New(ByVal transitionTypeId As Integer, ByVal lineColor As Color)

        MyBase.New(lineColor)
        Me.m_TransitionTypeId = transitionTypeId

    End Sub

    Friend ReadOnly Property TransitionTypeId() As Integer
        Get
            Return Me.m_TransitionTypeId
        End Get
    End Property

    Friend ReadOnly Property TransitionGroups() As List(Of Integer)
        Get
            Return Me.m_TransitionGroups
        End Get
    End Property

End Class

File: Diagram/ProbabilisticTransitionLine.cs
// ST-Sim: A SyncroSim Module for the ST-Sim State-and-Transition Model.
// Copyright © 2007-2018 Apex Resource Management Solution Ltd. (ApexRMS). All rights reserved.

using System.Drawing;
using System.Collections.Generic;

namespace SyncroSim.STSim
{
    internal class ProbabilisticTransitionLine : TransitionDiagramLine
    {
        private int m_TransitionTypeId;
        private List<int> m_TransitionGroups = new List<int>();

        public ProbabilisticTransitionLine(int transitionTypeId, Color lineColor) : base(lineColor)
        {
            this.m_TransitionTypeId = transitionTypeId;
        }

        internal int TransitionTypeId
        {
            get
            {
                return this.m_TransitionTypeId;
            }
        }

        internal List<int> TransitionGroups
        {
            get
            {
                return this.m_TransitionGroups;
            }
        }
    }
}
